diff --git src/cpp/CMakeLists.txt src/cpp/CMakeLists.txt
index 19029ef..0f618cf 100644
--- src/cpp/CMakeLists.txt
+++ src/cpp/CMakeLists.txt
@@ -76,7 +76,7 @@ if(MSVC OR MSVC_IDE)
     endif()
 else()
     #Create library
-    add_library(${PROJECT_NAME} ${${PROJECT_NAME}_SOURCES})
+    add_library(${PROJECT_NAME} STATIC ${${PROJECT_NAME}_SOURCES})
     set_target_properties(${PROJECT_NAME} PROPERTIES VERSION ${PROJECT_VERSION})
     set_target_properties(${PROJECT_NAME} PROPERTIES SOVERSION ${PROJECT_MAJOR_VERSION})
 
